### PR TITLE
Update GitHub Action Versions

### PR DESCRIPTION
**Pull request by wojciechcloudkubed**
# Resolves [#3847](https://github.com/microsoft/AzureTRE/issues/3847)
## What is being addressed

Update GitHub Action Versions to the latest ones 

## Acceptance criteria

- [ ] All actions are updated to their latest version

### CODE REVIEW

**Comment by tamirkamara** [comment]
@wojciechcloudkubed  There's no point in running a test deployment cycle (like with the / test command) when you update the pr_bot or the tre_deploy* workflows as those run from the main branch.

**Comment by wojciechcloudkubed** [comment]
> @wojciechcloudkubed There's no point in running a test deployment cycle (like with the / test command) when you update the pr_bot or the tre_deploy* workflows as those run from the main branch.

What would be the best approach to test it? @tamirkamara

File: .github/workflows/build_docker_images.yml [hunk 1]
```diff
@@ -1,11 +1,11 @@
 ---
 name: Docker build
 
-on:  # yamllint disable-line rule:truthy
+on: # yamllint disable-line rule:truthy
   pull_request:
     branches:
       - main
-      - 'feature/**'
```
[the remark this answers is not in the document]
Comment (marrobi) [reply .github/workflows/build_docker_images.yml:8]: Github Actions are already checked here - https://github.com/microsoft/AzureTRE/blob/ddddbf6e3e8d13db477f70dc2ca8048dd12c563a/.github/workflows/build_validation_develop.yml#L87

**Review by marrobi — APPROVED** [review]
LGTM - suggest test it on a fork before merging.

**Pull request closed without merging**